examples/agent/websocket/arbitrary_keys/main.go

```diff
@@ -268,9 +269,14 @@ func main() {
 	tOptions.Agent.Think.Prompt = "You are a helpful AI assistant."
 	tOptions.Agent.Listen.Provider["type"] = "deepgram"
 	tOptions.Agent.Listen.Provider["model"] = "nova-3"
-	tOptions.Agent.Speak.Provider["type"] = "deepgram"
-	tOptions.Agent.Speak.Provider["model"] = "aura-2-thalia-en"
-	tOptions.Agent.Speak.Provider["arbitrary_key"] = "test"
+	// Set speak provider with arbitrary key (single provider)
+	tOptions.Agent.Speak = interfacesv1.Speak{
+		Provider: map[string]interface{}{
+			"type":          "deepgram",
+			"model":         "aura-2-thalia-en",
+			"arbitrary_key": "test",
+		},
+	}
 	tOptions.Agent.Language = "en"
 	tOptions.Agent.Greeting = "Hello! How can I help you today?"
```

Comment:
Can we get an example for both of these uses? One is replacing the other, hinting at a breaking change as opposed to an additive change.